docs/reference/mapping/types/runtime.asciidoc

```diff
@@ -0,0 +1,585 @@
+      "type": "keyword",
+      "script": {
+        "source": "emit(doc['@timestamp'].value.dayOfWeekEnum.getDisplayName(TextStyle.FULL, Locale.ROOT))"
+      }
+    }
+  },
+  "aggs": {
+    "day_of_week": {
+      "terms": {
+        "field": "day_of_week"
+      }
+    }
+  }
+}
+----
+// TEST[continued]
+
+[[runtime-overriding-fields]]
+===== Overriding fields in the search request
+You can use the `runtime_mappings` section of the `_search` request to override
+field values inside objects by naming the runtime fields with dot notation. For
+example, index the following documents into `my-index`.
+
+[source, console]
+----
+POST my-index/_bulk?refresh=true
+{"index":{}}
+{"name":{"first":"Jose","last":"Hickman"}}
+{"index":{}}
+{"name":{"first":"India","last":"Avila","suffix":"II"}}
+----
+
+In the `runtime_mappings` section of the `_search` request, you can define a
+script that operates on the `name.first` field. The values you specify will
+override fields in `_source` for the duration of the query without modifying
+the field itself.
+
+The following request evaluates the `name.last` field and overrides the
+`name.first` field in the search request based on the script valuation.
+
+[source,console]
+----
+POST my-index/_search
+{
+  "runtime_mappings": {
+    "name.first": {
+      "type": "keyword",
+      "script": {
+        "source": "if (\"Hickman\".equals(doc[\"name.last.keyword\"].value)) {emit(\"Guile\");} else if (\"Avila\".equals(doc[\"name.last.keyword\"].value)) {emit(\"Anastasia\");}"
+      }
+    }
+  },
+  "query": {
+    "match": {
+      "name.first": "Anastasia"
+    }
+  }
+}
+----
+
+[[runtime-params]]
+==== Parameters for runtime fields
+Runtime fields accept the following parameters:
+
+[[runtime-params-type]]
+`type`::
+The type of runtime computation to perform at query time. Currently, runtime
+fields only support the `runtime` data type.
+
+[[runtime-params-runtime-type]]
+`runtime_type`::
+The <<mapping-types,field type>> for each scripted field. {es}
+supports `boolean`, `date`, `double`, `ip`, `keyword`, and `long`.
++
+Runtime fields with a `runtime_type` of `date` can accept the
+<<mapping-date-format,`format`>> parameter exactly as the `date` field type.
+
+[[runtime-params-script]]
+`script`::
+The <<modules-scripting-using,painless script>> that is evaluated at search
+time to produce the value of the runtime field.
+
+[[runtime-retrieving-fields]]
+==== Retrieving a runtime field
+Use the <<search-fields,`fields`>> parameter on the `_search` API to retrieve
+the values of runtime fields. Runtime fields won't display in `_source`, but
+the `fields` API works for all fields, even those that were not sent as part of
+the original `_source`.
+
+IMPORTANT: Queries against runtime fields are considered expensive. If
+<<query-dsl-allow-expensive-queries,`search.allow_expensive_queries`>> is set
+to `false`, expensive queries are not allowed and {es} will reject any queries
+against runtime fields.
```

Comment:
Maybe this should merge with the warning about async search?

Reply:
Also, it feels like it is in the wrong section. Retrieving the values isn't expensive. I mean, more expensive than `_source` access, but not terrible. But querying them can be.

Reply:
I'm moving this note to a **Tradeoffs** section and combining it with other relevant information.